haphilipsjs/__init__.py

```diff
@@ -747,13 +747,21 @@ def _decodeSystem(self, system) -> SystemType:
         return cast(SystemType, result)
 
     async def getSystem(self):
-        r = cast(Optional[SystemType], await self.getReq("system"))
-        if r:
-            self.system = self._decodeSystem(r)
-            self.name = r.get("name")
-        else:
-            self.system = {}
-            self.name = None
+        # Newest TV software requires https for system-info. Therefore we will try both protocols.
+        protocols = ["http", "https"]
+        orig_protocol = self.protocol
+        
+        for prot in protocols:
+            self.protocol = prot
```

Comment:
Please add and optional parameter to getReq for protocol (and forward to _url() function, then we don't need this back and forth.

Reply:
Also it should start with currently set protocol and fallback.

Reply:
> 

Added the getReq paramter in [717445c](https://github.com/danielperna84/ha-philipsjs/pull/33/commits/717445c8367dbbbee1b4b1fe85895fc71faa8307)
The other will follow the next days.

Reply:
Starts with currently set protocol now and falls back afterward.
Also the protocol is not set globally by the getSystem function anymore.